### DISCUSSION TITLE
Failure screenshot in base64 images

### DISCUSSION THREAD

**Discussion by thomasanup** [getgauge/gauge#2118] [General]
I am getting the failure screenshot as base64 embedded in html report .

Is there a setting where I can change this behaviour to store the screenshots in images folder . 

csharp (0.10.6)
html-report (4.0.6)
java (0.7.3)
ruby (0.5.2)
screenshot (0.0.1)
xml-report (0.2.2)

**Comment by zabil** [comment]
This feature is only available in the latest version of Gauge and Gauge html report. Please try to upgrade your installation to the latest versions. 

If you are using a custom screenshot handler please modify that to use the new api
 https://docs.gauge.org/writing-specifications.html?os=macos&language=javascript&ide=vscode#taking-custom-screenshots

**Reply by dhanushng88** [reply]
@zabil Even after using CustomScreenshot function that's given in the gauge documentation we are unable to see the screenshots in the html reports. Can you please advise us what's the issue and how to fix it?

in code we have used this:
gauge.customScreenshotWriter = async function () {
    const screenshotFilePath = path.join(process.env['gauge_screenshots_dir'], `screenshot-${process.hrtime.bigint()}.png`);
    await screenshot({ path: screenshotFilePath });
    return path.basename(screenshotFilePath);
};

afterStep(async () => {
    if (process.env.api_test_run === 'false') {
        gauge.screenshot();
    }
});

Gauge version: 1.2.1
Commit Hash: fa7918b

Plugins
-------
html-report (4.0.12)
js (2.3.17)
json-report (0.3.6)
screenshot (0.0.1)
xml-report (0.2.3)
[NoScreenShotsPresent.pdf](https://github.com/getgauge/gauge/files/7283209/NoScreenShotsPresent.pdf)

**Reply by sriv** [reply]
can you test this with latest version of gauge? You are on an old version, gauge 1.4.2 is the current released versino.
